File: src/gateway/containers/driver/src/modules/utils/logger.ts
type Severity = 'INFO' | 'WARN' | 'ERROR' | 'DEBUG';

export class Logger {

    static info(message: string): void {
        Logger.log(message, 'INFO');
      }
    
      static warn(message: string): void {
        Logger.log(message, 'WARN');
      }
    
      static error(message: string): void {
        Logger.log(message, 'ERROR');
      }

      static debug(message: string): void {
        Logger.log(message, 'DEBUG');
      }
    

    private static log(message: string, severity : Severity): void {
      const timestamp = new Date().toISOString();
      console.log(`[${timestamp}] ${severity} - ${message}`); 
      //  warning: javascript is single-thread, massive logging provides a memory intensive operations
    }
  
  }

File: src/gateway/containers/model/src/utils/logger.py
import logging
from datetime import datetime

class Logger:
    INFO = 'INFO'
    WARN = 'WARN'
    ERROR = 'ERROR'
    DEBUG = 'DEBUG'

    @staticmethod
    def info(message: str) -> None:
        Logger.log(message, Logger.INFO)

    @staticmethod
    def warn(message: str) -> None:
        Logger.log(message, Logger.WARN)

    @staticmethod
    def error(message: str) -> None:
        Logger.log(message, Logger.ERROR)

    @staticmethod
    def debug(message: str) -> None:
        Logger.log(message, Logger.DEBUG)

    @staticmethod
    def log(message: str, severity: str) -> None:
        timestamp = datetime.utcnow().isoformat()
        log_message = f"[{timestamp}] {severity} - {message}"
        print(log_message)

if __name__ == "__main__":
    Logger.info("This is an info message.")
    Logger.warn("This is a warning message.")
    Logger.error("This is an error message.")
    Logger.debug("This is a debug message.")
